### PR TITLE
Bug 1626746 - Push Health use Mozci to get Push parent

### PR DESCRIPTION
**Pull request by camd**
This introduces `mozci` to Push Health.  I'm only using it to get the parent information at this point.

### CODE REVIEW

**Comment by camd** [comment]
Hi Armen--  I hope I did this right with the caching and all with `adr`.  If I didn't, please let me know.  Thanks!

File: docker/entrypoint.sh [hunk 1]
```diff
@@ -5,6 +5,9 @@
 # Make non-zero exit codes & other errors fatal.
 set -euo pipefail
 
+mkdir -p /root/.config/adr
+cp docker/config.toml /root/.config/adr/config.toml
```
Comment (armenzg) [inline docker/entrypoint.sh:9]: You can do these directly in the `Dockerfile`.
You can instead set an `ADR_CONFIG_PATH` env variable to point to `/app/docker/config.toml`.
I find it more clear than using the default path that adr chooses.

**Comment by armenzg** [comment]
Also try to see why the Travis failures.

File: docker/config.toml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[adr]
+verbose = 1
+
+[adr.cache]
+retention = 40320
```
[the remark this answers is not in the document]
Comment (camd) [reply docker/config.toml]: Good point.  I actually modified that value in accordance with the `mozci` docs.  https://github.com/mozilla/mozci/blob/9794b21b6b50b939a7692bab68eae23e6bbbb9fe/docs/usage.rst

File: docker/entrypoint.sh (hunk 1 above)
Comment (camd) [reply docker/entrypoint.sh]: Yeah, that's better.  I did that instead.  Thanks!

**Comment by armenzg** [comment]
@camd you can use the same technique I used in here:
https://github.com/mozilla/mozci/commit/084ebb50e9f424926a1495a29e2f5e4d92187804#diff-7eb47fc18d4a6196f0940784484a5bf3R23

We create an adr_test_cache at the topsrcdir of Treeherder.

File: tests/push_health/test_compare.py [hunk 3]
```diff
@@ -1,189 +1,82 @@
+import pytest
 import datetime
 
 import responses
 
-from treeherder.model.models import Push
-from treeherder.push_health.compare import get_commit_history, get_response_object
+from treeherder.model.models import Push, Repository
```
Comment (camd) [inline tests/push_health/test_compare.py:6]: Since `mozci` does most of this work, we don't need all these test cases anymore.

**Pull request merged**